rolediscordbot.py

```diff
@@ -0,0 +1,110 @@
+    role_selection_channel = discord.utils.get(client.guilds[1].text_channels, name="roles-selection")
+
+    # Checking for channel
+    if role_selection_channel:
+        # Check if there's an existing role selection message
+        async for message in role_selection_channel.history(limit=1):
+            if message.author == client.user:
+                role_selection_message = message
+                break
+        
+        if role_selection_message:
+            print(f'Found existing role selection message: {role_selection_message.jump_url}')
+        else:
+            # Create a new role selection embed message
+            embed = discord.Embed(
+                title="Role Selection:",
+                description="React to any of the roles to get the one you want.",
+                color=discord.Color.blue()
+            )
+
+            # Custom emojis id's and role names dictionary
+            emoji_to_role = {
+                '<:comp:1173517373278523412>': 'Computer Role',
+                '<:linux:1173517388680011776>': 'Linux Role',
+                '<:py:1173518516125708308>': 'Python Role',
+            }
+
+            # Add fields to the embed message
+            for emoji, role_name in emoji_to_role.items():
+                role = discord.utils.get(role_selection_channel.guild.roles, name=role_name)
+                if role:
+                    embed.add_field(name=emoji, value=role.mention, inline=True)
+
+            # Send the embed message
+            role_selection_message = await role_selection_channel.send(embed=embed)
+
+            # Add reactions to the message
+            for emoji in emoji_to_role.keys():
+                await role_selection_message.add_reaction(emoji)
+
+            print(f'Created new role selection message: {role_selection_message.jump_url}')
+        
+        client.role_selection_message = role_selection_message
+
+# Calls when a reaction is added to the role selection message
+@client.event
+async def on_reaction_add(reaction, user):
+    if user.bot:
+        return
+    
+    # Check if the reaction is on the role selection message
+    if reaction.message == client.role_selection_message:
+        emoji_to_role = {
+            '<:comp:1173517373278523412>': 'Computer Role',
+            '<:linux:1173517388680011776>': 'Linux Role',
+            '<:py:1173518516125708308>': 'Python Role',
+        }
+
+        # Needed to convert emoji to string for custom emojis to compare with emoji_to_role dictionary keys
+        if str(reaction.emoji) in emoji_to_role:
+            role_name = emoji_to_role[str(reaction.emoji)]
+            role = discord.utils.get(user.guild.roles, name=role_name)
+            if role:
+                await user.add_roles(role)
+
+                if role_name == 'Computer Role':
+                    await user.send('You selected the Computer Role! You will be coding and gaining experience on how computers function!')
```

Comment:
I'd update these messages as well. I'll list them out for each role:

Autonomous Vehicle Design Teams:
- "You have been assigned the Autonomous Vehicle Design Team role! Here you can find the latest updates on our Autonomous Vehcile projects. If you are a member for this project, please reach out to Lorenz Carvajal or Alex Lyew to join your team's channels!"

Kotlin App:
- "You have been assigned the Kotlin App role! Here you can find the latest updates on our Android Mobile App project. If you are a member for this project, please reach out to Miguel Tejeda to join your team's channels!"

Swift App:
- "You have been assigned the Swift App role! Here you can find the latest updates on our iOS Mobile App project. If you are a member for this project, please reach out to Jesus Lopez to join your team's channels!"

Interview Prep:
- "You have been assigned the Interview Prep role! Here you can join us in our technical prep office hours, sign up for mock interviews and solve a LeetCode Question of the Week. If you have any questions, feel free to reach out to Mateo Slivka, Santiago Barrios or Diego Santos Gonzalez!"